lib/log_dna_batcher.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require "net/http"
+require "uri"
+require "json"
+require "socket"
+
+class LogDnaBatcher
+  def initialize(api_key, base_url: "https://logs.logdna.com/logs/ingest",
+    default_metadata: {"app" => $0, "pid" => Process.pid},
+    flush_interval: 0.5,
+    max_batch_size: 5)
+    @api_key = api_key
+    @flush_interval = flush_interval
+    @max_batch_size = max_batch_size
+    @input_queue = Queue.new
+    @http = nil
+    @base_url = URI.parse(base_url)
+    @default_metadata = default_metadata.dup.freeze
+
+    params = @base_url.query ? URI.decode_www_form(@base_url.query).to_h : {}
+    params["hostname"] = Socket.gethostname
+    params["app"] = $0
```

Comment:
You sure that actually works, when in the query parameters? Redundant with default_metadata, and I did default_metadata as it is because I seem to remember not being able to get "app" into the query string.

Reply:
Yep, you are right. Removed that `app` as param.